app/views/locations/_form.html.erb

```diff
@@ -0,0 +1,92 @@
+
+      <ul>
+        <% @location.errors.full_messages.each do |message| %>
+          <li><%= message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field">
+    <%= form.label :name %>
+    <%= form.text_field :name %>
+  </div>
+
+  <div class="field">
+    <%= form.label :details %>
+    <%= form.text_field :details %>
+  </div>
+
+  <div class="field">
+    <%= form.label :location_photo %>
+    <%= form.file_field :location_photo %>
+  </div>
+
+  <div class="field">
+    <%= form.label :location_latitude %>
+    <%= form.number_field :location_latitude, step: :any %>
+  </div>
+
+  <div class="field">
+    <%= form.label :location_longitude %>
+    <%= form.number_field :location_longitude, step: :any %>
+  </div>
+
+  <div class="field mb-3 opening-times-container">
+    <%= form.label :openingtimes, class: 'form-label' %>
+    <%= form.fields_for :openingtimes do |o| %>
+      <div class="field" id="opening-time-field-<%= o.object.id %>">
+        <%= o.select 'day',
+                     options_for_select({ 'Monday' => 0,
+                                          'Tuesday' => 1,
+                                          'Wednesday' => 2,
+                                          'Thursday' => 3,
+                                          'Friday' => 4,
+                                          'Saturday' => 5,
+                                          'Sunday' => 6
+                                        }, o.object.day),
+                     {},
+                     multiple: false %>
+        <%= o.time_field :opens, value: o.object.opens %>
+        <span>-</span>
+        <%= o.time_field :closes, value: o.object.closes %>
+        <%= link_to(openingtime_path(o.object.id), :method => :delete, :class => 'btn btn-mini',
+                    :remote => true, :onclick =>
+                      "{$('#opening-time-field-#{o.object.id}').next().remove();
+                       $('#opening-time-field-#{o.object.id}')[0].remove()}") do %>
+          <i class="fas fa-trash"></i>
+        <% end %>
+      </div>
+    <% end %>
+    <button class="btn" onclick="event.preventDefault(); addNewOpeningTime(this);"><i class="fas fa-plus"></i> Add
+      opening time
+    </button>
+  </div>
+
+  <div class="actions">
+    <%= form.submit %>
+  </div>
+
+  <script defer>
+      const addNewOpeningTime = (target) => {
+          $(`<div class="field" id="opening-time-field-new">
+           <select name="location[openingtimes_attributes][-1][day]" id="opening-time-field-new">
+                  <option value="0">Monday</option>
+                  <option value="1">Tuesday</option>
+                  <option value="2">Wednesday</option>
+                  <option value="3">Thursday</option>
+                  <option value="4">Friday</option>
+                  <option value="5">Saturday</option>
+                  <option value="6">Sunday</option></select>

```

Comment:
Possible to use method of OpeningTimes? Weird code duplication

Reply:
Same for people (and probably buildings, but UI for opening times is still missing?)

Reply:
We use the new number to string mapping method from the openingtime model now